### PR TITLE
Classifier: refactor tutorial components to use `useState` hooks

### LINKED ISSUE #2869: Popup tutorial opens and can’t be closed.
## Package
lib-classifier 

## Describe the bug
A volunteer is unable to work on new projects because the popup tutorial opens and is stuck on the first step. It can’t be closed. 

https://www.zooniverse.org/talk/17/2354191?comment=3863613&page=1

## To Reproduce
They’ve been able to reproduce this problem consistently on projects that use the new tutorial. See the Talk thread for details. 

Other volunteers don’t have problems, so reproducing this might require opening the tutorial with this volunteer’s project preferences loaded into the classifier.

EDIT: the problem goes away when the `clearurls` extension is disabled, so the bug seems to be triggered when project preferences can't be saved.

## Expected behavior
They should be able to dismiss the popup and use the classifier. 

## Screenshots
Screenshots are posted in the Talk thread.

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Refactor `ModalTutorial` to use `useState` for the popup's `active` state.
- Refactor `SlideTutorial` and `StepNavigation` to use `useState` to track the active tutorial step.
- Organise `TutorialStore` and `UserProjectPreferences` code into folders.
- Fix infinite recursion between the tutorial and UPP models when ETag headers are missing from API responses eg. when using a privacy plugin like ClearURLs.
- Fix `tutorials.setActiveTutorial` to check for a change in active tutorial before updating the seen time.
- Remove global actions and views that were used to track the popup tutorial's open/closed state and the active tutorial step.

With the changes here, it's possible to navigate and close the tutorial while using a privacy plugin, like ClearURLs. Tutorial seen status won't be recorded in user preferences, unless the plugin is disabled, but the classifier won't lock up on the first step of the tutorial while firing off an infinite loop of GET requests for the volunteer's UPP resource.

Package:
lib-classifier

Closes #2869.

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
#2869 is fully fixed now, by removing the tutorial store's `UPPObserver` and adding a check for changes in the active tutorial ID, before setting the tutorial seen time.

**Review by mcbouslog — APPROVED** [review]
Code refactors look great.

Tested signed-in and signed-out on Poets & Lovers and HMS NHS (on multiple workflows/tutorials within the project). Added `clearurls` extension in Firefox, confirmed issue, confirmed fix.

Tests look great. I'm reminded should revisit skipped tests across repo, but that's not related to this issue/PR.

**Review by mcbouslog — CHANGES_REQUESTED** [review]
Double-checking "Get started" from tutorial within task area, will update in a minute...

File: packages/lib-classifier/src/store/TutorialStore/TutorialStore.js [hunk 1]
```diff
@@ -30,12 +30,13 @@ const TutorialStore = types
       return self.loadingState !== asyncStates.success || (self.loadingState === asyncStates.success && !self.tutorial)
     },
 
-    get stepWithMedium () {
-      if (self.isActiveStepValid) {
-        const step = self.active.steps[self.activeStep]
-        return { step, medium: self.activeMedium }
+    stepWithMedium(index) {
+      const step = self.active.steps[index]
```
Comment (mcbouslog) [inline packages/lib-classifier/src/store/TutorialStore/TutorialStore.js:34]:
```suggestion
      const step = self.active?.steps[index]
```

ModalTutorial "Get Started" works, but SlideTutorial "Get Started" errors on this line trying to read `steps` with `self.active = undefined`.

**Review by mcbouslog — CHANGES_REQUESTED** [review]

File: packages/lib-classifier/src/store/TutorialStore/TutorialStore.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/store/TutorialStore/TutorialStore.js:34]: I think `self.active` here should also be wrapped in `tryReference()`. I'll double check.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/TutorialStore/TutorialStore.js:34]: That error should fail automated tests too. We shouldn’t be relying on manual testing to find it.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/TutorialStore/TutorialStore.js:34]: The Get Started button is clearing the active tutorial, which will crash the tutorial tab. I've removed `setActiveTutorial()` here:
https://github.com/zooniverse/front-end-monorepo/blob/928ef443ba9b2a1e700fa4a56239c6713e93e6b2/packages/lib-classifier/src/components/Classifier/components/TaskArea/TaskArea.js#L55-L58

**Review by mcbouslog — APPROVED** [review]
Recent changes LGTM!

**Comment by eatyourgreens** [comment]
Let's get this out on to staging.

**Pull request merged**